modules/openapi-generator/src/main/resources/go-server/model.mustache

```diff
@@ -36,6 +45,35 @@ type {{classname}} struct {
 {{/vars}}
 }{{/isEnum}}
 
+// UnmarshalJSON parse JSON while respecting the default values specified
+func (o *{{classname}}) UnmarshalJSON(data []byte) error {
+    type Alias {{classname}} // Avoid infinite recursion
+    aux := Alias{
+		{{#vars}}
+		{{#defaultValue}}
+		{{^isArray}}
+		{{#isBoolean}}
+		{{name}}: {{{.}}},
+		{{/isBoolean}}
+		{{#isNumeric}}
+		{{name}}: {{{.}}},
+		{{/isNumeric}}
+		{{^isBoolean}}
+		{{^isNumeric}}
+		{{name}}: "{{{.}}}",
+		{{/isNumeric}}
+		{{/isBoolean}}
+		{{/isArray}}
+		{{/defaultValue}}
+		{{/vars}}
+	}
+    if err := json.Unmarshal(data, &aux); err != nil {
+        return err
+    }
+    *o = {{classname}}(aux)
+    return nil
+}
```

Comment:
```suggestion
// UnmarshalJSON sets *m to a copy of data while respecting defaults if specified.
func (o *{{classname}}) UnmarshalJSON(data []byte) error {
	{{#vars}}
	{{#defaultValue}}
	{{^isArray}}
	{{#isBoolean}}
	m.{{name}} = {{{.}}}
	{{/isBoolean}}
	{{#isNumeric}}
	m.{{name}} = {{{.}}}
	{{/isNumeric}}
	{{^isBoolean}}
	{{^isNumeric}}
	m.{{name}} = "{{{.}}}"
	{{/isNumeric}}
	{{/isBoolean}}
	{{/isArray}}
	{{/defaultValue}}
	{{/vars}}
	type Alias {{classname}} // To avoid infinite recursion
    return json.Unmarshal(data, (*alias)(m))
}
```

I saw this cleaner code somewhere, very nice. Let's keep the func comments close to the original.